Dungeoneer/src/com/interrupt/dungeoneer/ui/layout/PlayerStatusEffects.java

```diff
@@ -0,0 +1,110 @@
+package com.interrupt.dungeoneer.ui.layout;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map.Entry;
+
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.scenes.scene2d.Actor;
+import com.badlogic.gdx.scenes.scene2d.ui.Label;
+import com.badlogic.gdx.scenes.scene2d.ui.VerticalGroup;
+import com.badlogic.gdx.utils.Align;
+import com.badlogic.gdx.utils.Array;
+import com.interrupt.dungeoneer.game.Game;
+import com.interrupt.dungeoneer.statuseffects.StatusEffect;
+import com.interrupt.dungeoneer.ui.UiSkin;
+
+public class PlayerStatusEffects extends Element {
+    HashMap<StatusEffect, Label> children = new HashMap<>();
+
+    @Override
+    protected Actor createActor() {
+        PlayerStatusEffects self = this;
+
+        VerticalGroup group = new VerticalGroup() {
+            @Override
+            public void act(float delta) {
+                super.act(delta);
+
+                Array<StatusEffect> statusEffects = Game.instance.player.statusEffects;
+
+                // Create new status effect lables
+                if (statusEffects != null) {
+                    for (int i = 0; i < statusEffects.size; i++) {
+                        final StatusEffect statusEffect = statusEffects.get(i);
+
+                        if (children.get(statusEffect) != null) {
+                            continue;
+                        }
+
+                        Label label = createLabel(statusEffect);
+                        addActor(label);
+                        children.put(statusEffect, label);
+                    }
+                }
+
+                // Remove old status effect labels
+                if (children.size() > 0) {
+                    Iterator<Entry<StatusEffect, Label>> iterator = children.entrySet().iterator();
+
+                    while (iterator.hasNext()) {
+                        Entry<StatusEffect, Label> child = iterator.next();
+
+                        if (statusEffects == null || !statusEffects.contains(child.getKey(), true)) {
+                            child.getValue().remove();
+                            iterator.remove();
+                        }
+                    }
+                }
```

Comment:
Potentially the child could remove itself? Depends how specialized we want the status effect label to be.

Reply:
True, but then I would need to pass in the parent to the child, so that the parent can clean-up the dictionary?

Reply:
Could you not just call it?
```java
self.children.remove(this.statusEffect);
```